catalog/content/core/content-core-videothumbnailplugin/src/main/java/ddf/content/plugin/video/VideoThumbnailPlugin.java

```diff
@@ -0,0 +1,242 @@
+                copyFFmpegBinary(inputStream);
+            }
+        } else {
+            throw new RuntimeException(
+                    "OS is not Linux, Mac, or Windows. No FFmpeg binary is available for this OS, so the plugin will not work.");
+        }
+
+        limitFFmpegProcessesSemaphore = new Semaphore(MAX_FFMPEG_PROCESSES, true);
+    }
+
+    /**
+     * Deletes the directory that holds the FFmpeg binary.
+     * <p>
+     * Called by Blueprint.
+     */
+    public void destroy() {
+        if (ffmpegPath != null) {
+            FileUtils.deleteQuietly(new File(FilenameUtils.getFullPathNoEndSeparator(ffmpegPath)));
+        }
+    }
+
+    private void copyFFmpegBinary(final InputStream inputStream) throws IOException {
+        final File ffmpegBinary = new File(ffmpegPath);
+
+        if (!ffmpegBinary.exists()) {
+            FileUtils.copyInputStreamToFile(inputStream, ffmpegBinary);
+            if (!ffmpegBinary.setExecutable(true)) {
+                LOGGER.warn(
+                        "Couldn't make FFmpeg binary at {} executable. It must be executable by its owner for the plugin to work.",
+                        ffmpegPath);
+            }
+        }
+    }
+
+    @Override
+    public CreateResponse process(final CreateResponse input) throws PluginExecutionException {
+        // TODO: How to handle application/octet-stream?
+        if (!input.getCreatedContentItem()
+                .getMimeType()
+                .getPrimaryType()
+                .equalsIgnoreCase("video")) {
+            return input;
+        }
+
+        final CreateRequest request = input.getRequest();
+        final Map<String, Serializable> properties = request.getProperties();
+        final ContentItem contentItem = input.getCreatedContentItem();
+
+        createThumbnail(contentItem, properties);
+
+        return new CreateResponseImpl(request,
+                contentItem,
+                input.getResponseProperties(),
+                properties);
+    }
+
+    @Override
+    public UpdateResponse process(final UpdateResponse input) throws PluginExecutionException {
+        // TODO: How to handle application/octet-stream?
+        if (!input.getUpdatedContentItem()
+                .getMimeType()
+                .getPrimaryType()
+                .equalsIgnoreCase("video")) {
+            return input;
+        }
+
+        final UpdateRequest request = input.getRequest();
+        final Map<String, Serializable> properties = request.getProperties();
+        final ContentItem contentItem = input.getUpdatedContentItem();
+
+        createThumbnail(contentItem, properties);
+
+        return new UpdateResponseImpl(request,
+                contentItem,
+                input.getResponseProperties(),
+                properties);
+    }
+
+    private CommandLine getFFmpegCommand(final String videoFilePath,
+            final String thumbnailFilePath) {
+        final String suppressPrintingBannerFlag = "-hide_banner";
```

Comment:
I think the only one worth configuring is `videoFramesToOutput`, i.e., how many thumbnails are generated. That would only be useful if we supported multiple thumbnails. I suppose the scale could be configured as well, but we aren't doing that right now with images in `TikaInputTransformer` (which is where I got 200 for the size).